kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/MountedResourceConfigProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.nio.file.Path;
+import java.util.function.BiFunction;
+
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.fabric8.kubernetes.api.model.VolumeMountBuilder;
+
+/**
+ * A {@link SecureConfigProvider} for Kubernetes Secrets and ConfigMaps
+ */
+public class MountedResourceConfigProvider implements SecureConfigProvider {
+
+    static final MountedResourceConfigProvider SECRET_PROVIDER = new MountedResourceConfigProvider("", "secrets",
+            (vb, resourceName) -> vb.withNewSecret().withSecretName(resourceName).endSecret());
+    static final MountedResourceConfigProvider CONFIGMAP_PROVIDER = new MountedResourceConfigProvider("", "configmaps",
```

Comment:
Did you think about enumerating the [items](https://kubernetes.io/docs/concepts/storage/volumes/#configmap:~:text=%3A%20log%2Dconfig-,items%3A,-%2D%20key%3A)?   I mean, so we have the key name, we _could_ bake that into the volume. 

That would mean if the key didn't exist in the configmap, the proxy would fail to start (fail early=better?).   On the other hand, the way you have allows the possibility for the key not to exist.  It's then be up to operand to handle the possibility of the file not existing (which it may or may not cope with).  I did wonder about overloading the placeholder with a way to gesture `optional` - but maybe that's a step too far.